apps/web/superuser_utils.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from datetime import datetime, timedelta
+
+from django.utils import timezone
+
+logger = logging.getLogger("audit")
+
+EXPIRY = 60 * 30  # 30 minutes
+
+MAX_CONCURRENT_PRIVILEGES = 5
+
+
+def apply_temporary_superuser_access(request, grant):
+    if not isinstance(grant, str) or not grant.strip():
+        raise ValueError("Invalid grant")
+
+    remove_expired_temporary_superuser_access(request)
+    if has_temporary_superuser_access(request, grant):
+        return
+
+    elevated_privileges = request.session.get("elevated_privileges", [])
+    if len(elevated_privileges) >= MAX_CONCURRENT_PRIVILEGES:
```

Comment:
When we have 5 active "elevated privileges" and need to (for some reason) get a 6th one, we'd have to wait until one of the 5 expires first, yes?